yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -514,13 +582,100 @@ func (yb *TargetYugabyteDB) importBatch(conn *pgx.Conn, batch Batch, args *Impor
 		return res.RowsAffected(), err
 	}
 
-	err = yb.recordEntryInDB(tx, batch, res.RowsAffected())
+	// 5. Record the import in the DB.
+	err = yb.recordEntryInDB(conn, batch, res.RowsAffected())
 	if err != nil {
 		err = fmt.Errorf("record entry in DB for batch %q: %w", batch.GetFilePath(), err)
 	}
 	return res.RowsAffected(), err
 }
 
+// importBatchFastRecover is used to import a batch which was previously tried via fast path but failed
+func (yb *TargetYugabyteDB) importBatchFastRecover(conn *pgx.Conn, batch Batch, args *ImportBatchArgs) (int64, error) {
+	// 1. Check if the split is already imported.
+	alreadyImported, rowsAffected, err := yb.isBatchAlreadyImported(conn, batch)
+	if err != nil {
+		return 0, err
+	}
+	if alreadyImported {
+		return rowsAffected, nil
+	}
+
+	// 2. Open the batch file as datafile
+	df, err := batch.OpenAsDataFile()
+	if err != nil {
+		return 0, fmt.Errorf("open file %s: %w", batch.GetFilePath(), err)
+	}
+	defer df.Close()
+
+	// 3. Read the batch file line by line(row) and build INSERT statment for it based on PK conflict action
+	var rowsIgnored int64 = 0
+	rowsAffected = 0
+
+	// [TODO: Optional] Use PreparedStatement Caching on coonPool for better performance
+	insertStmt := args.GetInsertPreparedStmtForBatchImport()
+	for {
+		line, _, readLinErr := df.NextLine()
```

Comment:
is next line appropriate here? Some questions (i don't know the answers to these): 
Are there cases where a \n in the text field causes problems with next line?

I see in fileBatchProducer that we use `line, currentBytesRead, readLineErr = p.dataFile.NextLine()` , so I'm guessing it shouldn't be a problem.
cc: @priyanshi-yb

Reply:
yeah i just relied on NextLine() since we use it at other places to get the row